File: src/main/java/ru/spaceinvasion/mechanic/game/Field.java
package ru.spaceinvasion.mechanic.game;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import ru.spaceinvasion.mechanic.game.messages.GameMessage;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by egor on 17.11.17.
 */
public class Field extends GamePart {

    @NotNull
    private Player player1;
    @NotNull
    private Player player2;
    @Nullable
    private Unit unit1;
    @Nullable
    private Unit unit2;
    @NotNull
    private List<Tower> towers = new ArrayList<>();
    @NotNull
    private List<Shot> shots = new ArrayList<>();
    @NotNull
    private Boolean installedBombOn1 = false;
    @NotNull
    private Boolean installedBombOn2 = false;

    public Field(@NotNull Mediator mediator,
                 @NotNull Integer gamePartId,
                 @NotNull Player player1,
                 @NotNull Player player2) {
        super(mediator, gamePartId);
        this.player1 = player1;
        this.player2 = player2;
    }

    @Override
    public void notify(GameMessage message) {

    }
}

File: src/main/java/ru/spaceinvasion/mechanic/game/models/Field.kt
package ru.spaceinvasion.mechanic.game.models

import ru.spaceinvasion.mechanic.game.*
import ru.spaceinvasion.mechanic.game.messages.GameMessage
import ru.spaceinvasion.mechanic.game.messages.StartGameMessage
import ru.spaceinvasion.models.Coordinates
import ru.spaceinvasion.resources.Constants

import java.util.ArrayList

/**
 * Created by egor on 17.11.17.
 */
class Field(mediator: Mediator<GamePart>,
            gamePartId: Int) : GamePart(mediator, gamePartId) {
    private var unit1: Unit? = null
    private var unit2: Unit? = null
    private val towers = ArrayList<Tower>()
    private val shots = ArrayList<Shot>()
    private var installedBombOn1 = false
    private var installedBombOn2 = false

    override fun notify(message: GameMessage) {
        when (message.javaClass) {
            StartGameMessage::class.java-> {
                unit1 = Unit(mediator, message.messageId, Constants.COORDINATES_OF_UNIT1_START);
                unit1 = Unit(mediator, message.messageId, Constants.COORDINATES_OF_UNIT2_START);
                mediator.registerColleague(Unit::class.java, unit1!!)
                mediator.registerColleague(Unit::class.java, unit2!!)
            }
        }
    }
}
